### PR TITLE
Be volunteer form

### LINKED ISSUE #21: feat:  ❗ Finalize the neighbourhood data form with geolocation
This model will be opened when the user clicks 'Gönüllü Ol' button.

All fields are required:  
- `name` - text
- `surname` - text
- `phone` - number - masked if we can
- `email` - email
- `neighbourId` - number - hidden - selected neighbourhood from the drawer

add reCaptcha to the form.
display confirmation or rejected case on form submit

API Endpoint: ` POST /api/volunteer-form`

On Click of "KVKK Metni" text implement a modal showing "KVKK Metni"

BasvuruSecim
KVKK

** discord username: `@Mustafa Bolat#4560 `**

### PR DESCRIPTION
**Pull request by salihdev0**
# Description

- Volunteering application submission process has been completed.
- KVKK modal is defined and opens when KVKK text is opened or approved.
- Exiting the alert modal before submitting the form.
- Validation processes were performed with Formik and Yup.
- With React toastify, the notification is published according to the api response.

## Note

**discord username: ssi#9999**

**closes #21**

## Things to check before creating a PR
- [ ] I have inspected my topic, checked.
- [x] If it is a core feature, executed detailed tests.

## Creating PR rules
- [x] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [x] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [x] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [x] Related file selection: Only relevant files should be touched and no other files should be affected.
- [x] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [x] Clean commit history: The commits where the changes are made should be clear and organized.
- [x] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [x] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

### CODE REVIEW

File: src/components/Forms/BeVolunteer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+  useEffect(() => {
+    if (neighborhoodId) {
+      setFieldValue("neighborhoodId", neighborhoodId);
+    }
+  }, [setFieldValue, neighborhoodId]);
+
+  useEffect(() => {
+    if (form.values.kvkkAccepted) {
+      toggleKVKK();
+    }
```
Comment (usirin) [inline src/components/Forms/BeVolunteer/index.tsx:145]: bunu yapmamiz gerekiyo mu illa? gereklilik sanirim ama checkbox'i sectikten sonra baska bi modal acilmasini beklemiyordum. cok onemli degil ama emin olmak istedim.

**Review by usirin — APPROVED** [review]

File: src/components/Forms/BeVolunteer/index.tsx (hunk 1 above)
Comment (salihdev0) [reply src/components/Forms/BeVolunteer/index.tsx:145]: okumadan geçmesinler diye ekledim herhangi bir zorunluluğu yok kaldırabilirim

Comment (usirin) [reply src/components/Forms/BeVolunteer/index.tsx:145]: tamamdir sikinti yok, okundugundan emin olmak daha muhim bence de. backend duzeldiginde mergeleyebilirsin abi.

**Comment by eraygundogmus** [comment]
This feature is cancelled because organizational reasons. Please reach us on Discord for more detail, thank you for your contribution!

**Pull request closed without merging**